src/common/logger.cpp

```diff
@@ -13,10 +13,42 @@
 #include "mpi.h"
 
 namespace {
+/**
+ * The actual signal handler - must have this exact signature
+ * MPI is probably dead by this point, SLIC is definitely dead
+ */
 void signalHandler(int signal)
 {
-  // Will clean up MPI via call to axom::Utilities::processAbort()
-  SLIC_ERROR("Received signal " << signal << ", exiting");
+  int mpi_finalized;
+  MPI_Finalized(&mpi_finalized);
+  if (!mpi_finalized) {
+    MPI_Finalize();
+  }
+  // This works, but is it UB to call MPI_Comm_rank here?
+  int rank;
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
```

Comment:
I can't imagine calling MPI_Comm_rank after MPI_Finalized is safe or has expected behavior.

Reply:
Yeah, I just found the relevant part of the standard/manual.  I think the fact that MPI is dead by the time the signal handler is called means that having a signal handler in the first place isn't particularly useful.  SCR won't be of any help without MPI so I think all a "stranded" process could do is dump its local data and hope that it can be stitched together on restart.